src/autopas/utils/ParticleCellHelpers.h

```diff
@@ -38,6 +38,7 @@ static bool checkParticleInCellAndUpdateByID(CellType &cell, const typename Cell
 template <class CellType>
 static bool checkParticleInCellAndUpdateByIDAndPosition(CellType &cell, const typename CellType::ParticleType &particle,
                                                         double absError) {
+  std::lock_guard<AutoPasLock> cellLock(cell._cellLock);
```

Comment:
Do we need the lock here?
If so, I guess also in the function `checkParticleInCellAndUpdateByID()`